tests/unit/lms/services/grouping_test.py

```diff
@@ -205,9 +205,10 @@ def test_get_course_grouping_for_user_with_different_user(
         assert not groupings
 
     @pytest.fixture
-    def with_course_memberships(self, svc, application_instance):
+    def with_course_memberships(self, svc, db_session, application_instance):
         courses = factories.Course.create_batch(5)
         user = factories.User(application_instance=application_instance)
+        db_session.flush()
```

Comment:
This seems to be a staple of my PR now.

I'd say this worked before because adding the course and user to the session (via  `GroupingMembership(grouping=group, user=user)` populated the IDs. Now using the core API we need the IDs beforehand.